src/main/java/online/bookstore/controller/BookController.java

```diff
@@ -26,32 +27,38 @@ public class BookController {
     private final BookService bookService;
 
     @GetMapping
+    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
```

Comment:
In your solution, the user-role has many-to-many relations. You give a USER role by default to each registered user, meaning that ADMIN also has a USER role. So you can use only the USER role here  
```suggestion
    @PreAuthorize("hasAnyRole('ROLE_USER')")
```